RELEASE.md

```diff
@@ -1,23 +1,233 @@
-2. Run `./scripts/prepare-release.sh X.Y.Z`
-    this script will prepare a release commit with the version bumped in appropriate places and also create a local vX.Y.Z git tag;
-    feel free to verify the commit with `git log -p -1` after the script was executed successfully
-3. Push the commit to scylladb/scylla-rust-driver `main` branch, e.g. `git push origin HEAD:refs/heads/main`
-4. Push the newly created tag to scylladb/scylla-rust-driver, e.g. `git push origin vX.Y.Z`
-5. Write release notes. You can find an example here: https://groups.google.com/g/scylladb-users/c/uOfmdTeq6qM or here: https://github.com/scylladb/scylla-rust-driver/releases/tag/v0.5.0
-6. Go to https://github.com/scylladb/scylla-rust-driver/releases , click the `Draft new release` button and follow the procedure to create a new release on GitHub. Use the release notes as its description.
-7. Publish the crate(s) on crates.io. For `scylla` crate, go directly to the crate directory, i.e. `/your/repo/path/scylla-rust-driver/scylla`, and run `cargo publish`. It will ask you for an access token, follow the instructions if you haven't set one up.
-8. Send the release notes to scylladb-users (https://groups.google.com/g/scylladb-users) as well as Cassandra users (https://lists.apache.org/list.html?user@cassandra.apache.org) mailing lists.
-9. Publish the documentation for the new version. In `docs/source/conf.py`,
-update the following:
-- `TAGS = [' ']` - add the new release tag to the list
-- `LATEST_VERSION = ' '` - replace the previous tag with the new release tag
+# Maintenance
+
+The following document includes information how to release scylla-rust-driver and
+other informations / procedures useful for maintainers.
+
+## Crates and versions
+
+This workspace consists of 4 crates:
+- `scylla`: main driver crate to be used by users. Depends on `scylla-cql` and `scylla-macros`.
+- `scylla-macros`: provides derive macros for users to be able to use their structs as UDTs / query parameters.
+- `scylla-cql`: low-level implementation of CQL binary protocol. Depends on `scylla-macros`.
+- `scylla-proxy`: mostly internal crate that aids testing. Depends on `scylla-cql`.
+
+We publish all 4 crates on [crates.io](https://crates.io/crates/scylla).
+
+Versions of the crates are decided with strict accordance to semver.
+For a sources on semver and semver in rust see:
+- https://doc.rust-lang.org/cargo/reference/resolver.html
+- https://doc.rust-lang.org/cargo/reference/semver.html
+- https://semver.org/
+- https://rust-lang.github.io/api-guidelines/necessities.html#public-dependencies-of-a-stable-crate-are-stable-c-stable (and some other parts of this book).
+
+What needs to be considered for releasing:
+- You can't change major version of scylla-cql / scylla-macros dependency in a minor update of scylla crate.
+For example, if scylla 0.13.0 depends on scylla-cql 0.4, then scylla 0.13.1 can't depend on scylla-cql 0.5.
```

Comment:
Subtlety: `can't` -> `mustn't`, to stress that it's the releaser's responsibility to ensure that.